backend/src/nodes/impl/upscale/convenient_upscale.py

```diff
@@ -56,23 +57,33 @@ def convenient_upscale(
             unique_alpha = np.full(rgb.shape[:-1], unique[0], np.float32)
             return np.dstack((rgb, unique_alpha))
 
-        # Transparency hack (white/black background difference alpha)
-        black, white = with_black_and_white_backgrounds(img)
-        black_up = as_target_channels(
-            upscale(as_target_channels(black, model_in_nc, True)), 3, True
-        )
-        white_up = as_target_channels(
-            upscale(as_target_channels(white, model_in_nc, True)), 3, True
-        )
-
-        # Interpolate between the alpha values to get a more defined alpha
-        alpha_candidates = 1 - (white_up - black_up)  #  type: ignore
-        alpha_min = np.min(alpha_candidates, axis=2)
-        alpha_max = np.max(alpha_candidates, axis=2)
-        alpha_mean = np.mean(alpha_candidates, axis=2)
-        alpha = alpha_max * alpha_mean + alpha_min * (1 - alpha_mean)
-
-        return np.dstack((black_up, alpha))
+        if separate_alpha:
+            # Upscale the RGB channels and alpha channel separately
+            rgb = as_target_channels(
+                upscale(as_target_channels(img[:, :, :3], model_in_nc, True)), 3, True
+            )
```

Comment:
This has a problem. Since the RGB value classically isn't defined for alpha=0 (but typically set to 0), models will typically add a black edge to images with binary alpha. 

Solving this problem is the very purpose of my Fill Alpha node. By filling the transparent pixels with nearby colors, and therefor extending the image into the transparent pixels (inpainting them), models don't produce black edges anymore. The issue now is that the results depends on the inpaint algorithm. This is why the Fill Alpha node has multiple modes and the user has to choose one depending on the image.

So what do we do here? The current solution isn't really ideal, but the better solution requires additional user input and can already be achieved with today's nodes.

Reply:
> So what do we do here? The current solution isn't really ideal, but the better solution requires additional user input and can already be achieved with today's nodes.

I don't want this to scope creep into making the upscale nodes one monolithic node of every possible option. If it doesn't even make sense to do this, then I should probably just scrap this PR.

Reply:
True. Then let's not do any sort of inpainting right now. If black borders are an issue, then users can use our existing nodes to fix the issue.

Reply:
So let's do this feature.